modules/storages/app/controllers/storages/admin/storages_controller.rb

```diff
@@ -52,7 +52,12 @@ def index
 
   # Show page with details of one Storage object.
   # Called by: Global app/config/routes.rb to serve Web page
-  def show; end
+  def show
+    # Replace the client secret by **** for security reasons
+    if @object.oauth_client_secret.present?
+      @object.oauth_client_secret = "****"
```

Comment:
@fraber I am afraid that overwriting a model's attribute for the purpose of presentation in the view is not the best idea. Who knows that model is going to be used for next (in whatever hook) and then there is wrong data stored in it. So I'd prefer decoupling the presentation from the data here.

Reply:
Hi, this is all just temporary until Rails 7 introduces this new way of storing secrets.
I've added a check for length > 10 in the model to make sure the secret is not overwritten.

Reply:
Ok, I'll modify this and introduce a parameter in the view or something similar.